builtInFunctions/baseComponentsHolder.go

```diff
@@ -63,8 +69,59 @@ func getLatestEsdtData(currentEsdtData, transferEsdtData *esdt.ESDigitalToken, e
 	currentEsdtDataVersion := big.NewInt(0).SetBytes(currentEsdtData.Reserved).Uint64()
 	transferEsdtDataVersion := big.NewInt(0).SetBytes(transferEsdtData.Reserved).Uint64()
 
-	if currentEsdtDataVersion > transferEsdtDataVersion {
-		return currentEsdtData
+	if transferEsdtDataVersion >= currentEsdtDataVersion {
+		return transferEsdtData
```

Comment:
this is not the thing you should do. because some of the updates might change only one field, not all the other ones.

 I think we could use the Reserved field better - where we have list of information when one or the other field was updated.
 
 The thing we discussed, that even is the version is newer, we have to see whether there is a field with nil, and if yes, we should copy that from the other metaData.

But I think having a versioning structure in the reserved bytes - for each field, might be better.

Reply:
Will refactor as discussed.